engine/src/main/java/com/agiletec/aps/system/services/pagemodel/cache/PageModelManagerCacheWrapper.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2018-Present Entando Inc. (http://www.entando.com) All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+package com.agiletec.aps.system.services.pagemodel.cache;
+
+import com.agiletec.aps.system.exception.ApsSystemException;
+import com.agiletec.aps.system.services.pagemodel.IPageModelDAO;
+import com.agiletec.aps.system.services.pagemodel.PageModel;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.cache.Cache;
+import org.springframework.cache.CacheManager;
+
+/**
+ * @author E.Santoboni
+ */
+public class PageModelManagerCacheWrapper implements IPageModelManagerCacheWrapper {
+
+	private static final Logger _logger = LoggerFactory.getLogger(PageModelManagerCacheWrapper.class);
+
+	private CacheManager _springCacheManager;
+
+	@Override
+	public void loadPageModels(IPageModelDAO pageModelDAO) throws ApsSystemException {
+		try {
+			Cache cache = this.getCache();
+			List<String> codes = (List<String>) this.get(cache, PAGE_MODEL_CODES_CACHE_NAME, List.class);
+			if (null != codes) {
+				for (int i = 0; i < codes.size(); i++) {
+					String code = codes.get(i);
+					cache.evict(PAGE_MODEL_CACHE_NAME_PREFIX + code);

```

Comment:
Why evict on a load call? Is this strictly to clear and reload the cached PageModels?

Reply:
Because if you delete a page directly from the database (things that happen) and reload the configuration, if you do not delete the previous elements the cache would remain "dirty"